### PR TITLE
Upgrade streamlit version and match apis

### PR DESCRIPTION
**Pull request by sooahleex**
### Summary

- Upgrade streamlit version and do not set limit for version of it

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added the description of my changes into [CHANGELOG](https://github.com/openvinotoolkit/datumaro/blob/develop/CHANGELOG.md).​
- [ ] I have updated the [documentation](https://github.com/openvinotoolkit/datumaro/tree/develop/docs) accordingly

### License

- [ ] I submit _my code changes_ under the same [MIT License](https://github.com/openvinotoolkit/datumaro/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2024 Intel Corporation
#
# SPDX-License-Identifier: MIT
```

### CODE REVIEW

File: gui/requirements.txt [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Core
-streamlit~=1.29.0
+streamlit
```
Comment (yunchu) [inline gui/requirements.txt]:
```suggestion
streamlit==1.32.0
```
please pin the dependency with specific version

File: tests/requirements.txt [hunk 2]
```diff
@@ -9,7 +9,7 @@ coverage
 pytest-csv
 
 # DatumaroApp
-streamlit~=1.29.0
+streamlit
```
Comment (yunchu) [inline tests/requirements.txt]:
```suggestion
streamlit==1.32.0
```
same as above

File: gui/requirements.txt (hunk 1 above)
Comment (sooahleex) [reply gui/requirements.txt]: Fix in [3a5e361](https://github.com/openvinotoolkit/datumaro/pull/1334/commits/3a5e361550294097c4afe167f8028d919182c76d)

File: tests/requirements.txt (hunk 2 above)
Comment (sooahleex) [reply tests/requirements.txt]: Fix in [3a5e361](https://github.com/openvinotoolkit/datumaro/pull/1334/commits/3a5e361550294097c4afe167f8028d919182c76d)

**Review by yunchu — APPROVED** [review]

**Pull request merged**